TYPO3.Flow/Classes/TYPO3/Flow/Package/Package.php

```diff
@@ -456,15 +456,16 @@ public function getComposerManifest($key = null)
     }
 
     /**
-     * Get the installed package version (from composer)
+     * Get the installed package version (from composer) and as fallback the version given by package meta data.
      *
      * @return string
      * @api
      * TODO: Should be added to the interface in the next major Flow version (4.0)
      */
     public function getInstalledVersion()
     {
-        return PackageManager::getPackageVersion($this->composerName);
+        $installedVersion = PackageManager::getPackageVersion($this->composerName);
+        return  $installedVersion === '' ? $installedVersion : $this->getPackageMetaData()->getVersion();
```

Comment:
I'd go with

   return PackageManager::getPackageVersion($this->composerName) ?: $this->getPackageMetaData()->getVersion();